tyssue/collisions/solvers.py

```diff
@@ -40,6 +45,57 @@ def with_collision_correction(*args, **kwargs):
     return with_collision_correction
 
 
+def solve_self_intersect_face(eptm):
+    face_self_intersect = eptm.edge_df.groupby("face").apply(_do_face_self_intersect)
+
+    for f in face_self_intersect[face_self_intersect].index:
+        sorted_edge = np.array(_ordered_edges(
+            eptm.edge_df[eptm.edge_df['face'] == f][['srce', 'trgt', 'face']])).flatten()[3::4]
+        angle_list = np.arctan2(
+            eptm.edge_df.loc[sorted_edge]['sy'].to_numpy() - eptm.edge_df.loc[sorted_edge]['fy'].to_numpy(),
+            eptm.edge_df.loc[sorted_edge]['sx'].to_numpy() - eptm.edge_df.loc[sorted_edge][
+                'fx'].to_numpy()) * 180 / np.pi + 180
+
+        angle_e = pd.DataFrame(angle_list, index=sorted_edge, columns=['angle'])
+
+        if np.where(np.min(angle_e['angle']) == angle_e['angle'])[0] != 0:
+            pos_s = np.where(np.min(angle_e['angle']) == angle_e['angle'])[0][0]
+            angle_e = pd.concat([angle_e.iloc[pos_s:], angle_e.iloc[:pos_s]])
+
+        angle_e = pd.concat([angle_e, angle_e.iloc[[0]]])
+        angle_e.iloc[-1]['angle'] += 360
+
+        pos_s = np.where(angle_e.diff()['angle'] < 0)[0][0]
+        v1 = eptm.edge_df.loc[angle_e.index[pos_s]]['srce']
+        v2 = eptm.edge_df.loc[angle_e.index[pos_s - 1]]['srce']
+        if eptm.vert_df.loc[v1]['segment'] == eptm.vert_df.loc[v2]['segment']:
+            v1_x, v1_y = eptm.vert_df.loc[v1][['x', 'y']]
+            v2_x, v2_y = eptm.vert_df.loc[v2][['x', 'y']]
+            eptm.vert_df.loc[v1, ['x', 'y']] = v2_x, v2_y
+            eptm.vert_df.loc[v2, ['x', 'y']] = v1_x, v1_y
+
+
+def _do_face_self_intersect(edge):
+    sorted_edge = np.array(_ordered_edges(
+        edge[['srce', 'trgt', 'face']])).flatten()[3::4]
+    angle_list = np.arctan2(
+        edge.loc[sorted_edge]['sy'].to_numpy() - edge.loc[sorted_edge]['fy'].to_numpy(),
+        edge.loc[sorted_edge]['sx'].to_numpy() - edge.loc[sorted_edge][
+            'fx'].to_numpy()) * 180 / np.pi + 180
+
+    angle_e = pd.DataFrame(angle_list, index=sorted_edge, columns=['angle'])
+
+    if np.where(np.min(angle_e['angle']) == angle_e['angle'])[0] != 0:
+        pos_s = np.where(np.min(angle_e['angle']) == angle_e['angle'])[0][0]
+        angle_e = pd.concat([angle_e.iloc[pos_s:], angle_e.iloc[:pos_s]])
+
+    angle_e = pd.concat([angle_e, angle_e.iloc[[0]]])
+    angle_e.iloc[-1]['angle'] += 360
```

Comment:
`+= 2*np.pi`

Reply:
I think we can vectorize all that by sorting all the edges over the whole edge_df with `reset_index(ordered=True)` and then computing the angle everywhere?

Maybe I am not understanding something, but in your implementation, is there not a problem if a face is not convex?

![rect2401](https://user-images.githubusercontent.com/1426279/201949643-9d1c1e1f-a44e-4827-932c-eae9ca4cfd41.png)

Reply:
well I think this only limits us to "star convex" cells, which is not so bad (the cell we see here is not very physiological).
Maybe we can live with that for the moment (with a warning in the doc)?
But in this case I think your alog boils down to:
```python
angle = np.arctan2(
   sheet.edge_df["dy"],
   sheet.edge_df["dx"]
)
to_flip = sheet.edge_df[angle < 0, ["srce", "trgt"]]

(
   sheet.vert_df[to_flip["srce"].to_numpy(), ["x", "y"]], 
   sheet.vert_df[to_flip["trgt"].to_numpy(), ["x", "y"]] 
) = (
   sheet.vert_df[to_flip["trgt"].to_numpy(), ["x", "y"]], 
   sheet.vert_df[to_flip["srce"].to_numpy(), ["x", "y"]] 
)
```
As we just flip the negative angles (i.e. the red arrow on the drawing above)



```

Reply:
Well, in fact I have a very strange behavior if I use `reset_index(ordered=True)`: vertex stop moving. 
They are supposed to move because when I plot forces, I see big arrow forces for a few vertex. And if I apply `find_energy_min` without collisions, vertex moves again... 
I really don't know what append to "block" vertex movement... 
So for now I will keep the code that work, and focus on why sometime tissue is out of control.... :unamused: 
![image](https://user-images.githubusercontent.com/17041165/202240153-2524f096-f7d5-46ba-84ed-18ad3efb7339.png)